.github/workflows/release.yml

```diff
@@ -27,25 +27,29 @@ jobs:
       - name: Checkout the code
         uses: actions/checkout@v4
 
-      - name: Setup mamba
-        uses: mamba-org/setup-micromamba@v1
+      - name: Install uv
+        uses: astral-sh/setup-uv@v3
         with:
-          environment-file: env.yml
-          environment-name: my_env
-          cache-environment: true
-          cache-downloads: true
-          create-args: >-
-            pip
-            semver
-            python-build
-            setuptools_scm
+          enable-cache: true
+          cache-dependency-glob: ./pyproject.toml
+      
+      - name: Install the project
+        run: |
+          uv venv --python 3.12
+          uv sync --all-groups
+
+      - name: Install build dependencies
+        run: uv pip install python-build semver setuptools_scm
```

Comment:
**Comment**
Out of scope for this PR, but uv also has commands to [build](https://docs.astral.sh/uv/reference/cli/#uv-build) and [publish](https://docs.astral.sh/uv/reference/cli/#uv-publish) packages.